proposal.rst

```diff
@@ -267,8 +292,69 @@ Project Timeline
 .. If you will have other commitments during the project time (for example,
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
-
-FILLME
+In this summer I would like to do three things:
+
+    1. Make the go bindings compatible with the xapian build system.
+
+    2.  Provide bindings in go and complete API reference (as it is provided for C++)
+        using godoc,Go lang's official documentation tool where any one can look at the 
+        documentation if they just have go installed.The user can view the xapian 
+        documentation by just running go doc xapian from the terminal or command prompt.
+        And the documentation would be providied in godoc official website 
+        and xapian.org.
+        I have divided this part of work into to 10 weeks.
+
+    3.  Provide the go get feature for using xapian bindings in go with 
+        proper guide lines regarding whether the xapian core is installed
+        or not.
+
+
+Before May 18,2020 :
+I would finish up the build system part of work for go bindings that 
+are generated purely from xapian-headers.i and xapian-head.i such that
+these can be used like any other bindings and in any platform that other xapian bindings can be used in.
+
+Each week for first 3 days I would work on provding API , that is writing swig interface 
+for the functions and classes which aren't generated from 
+existing swig interfaces(xapian-headers.i and xapian-head.i) with typemaps and other features of swig in go.i.
+Next 2 days documenting the code with examples and the last day for testing.
+I have grouped different classes for each week.
+
+
+First Month : 
+
+  Week1 :
+        Work on Document , Registry , Query , Stem implementation , Stem classes.
+  Week2 :
+        Work on PositionIterators , PostingIterator , TermTterator , ValueIterators
+        
+  Week3 :
+        Work Eset , Mset , Rset , EsetIterator , MsetIterator classes.
+       
+  Week4 :
+        Work on MatchDecider , Enquire , ExapandDecider , ExapandDeciderAnd ,
+        ExapandDeciderFilterPrefix , ExapandDeciderFilterTerms , KeyMaker ,MultiValueKeyMaker ,
+        Stopper classes.
+     
+Second Month :
+  Week5 :
+        Work on all the Derived classes on RangeProcessor.
+  Week6 :
+        Work on all Weight classes.
+  Week7 :
+        Work on all the Derived classes of Posting Source.
+  Week8 :
+        Work on all the Derived classes of Matchspy classes and classes related to Latlong.
+
+Third Month : 
+  Week9 :
+        Work on Database class.
+  Week10 :
+        Work on Exceptions.
```

Comment:
The work in these weeks (9 & 10) are almost certainly more important than PostingSource, MatchSpy, and some of the others you have in the second month. However you might be able to split the work here — ensure enough of the Database functionality is working for most uses, then move on to things like MatchSpy before coming back to tidy up the rest of Database. The more you can get into this level of detail, the easier it will be to make a success of the project — so we expect quite a lot of detail in the proposal.